pkg/cryptoutil/keystore_message_utils_test.go

```diff
@@ -520,3 +522,78 @@ func TestMessageKeyHolderSubscription(t *testing.T) {
 		testMessageKeyHolderSubscription(t, testCase.expectedNewDevices, testCase.slow)
 	}
 }
+
+func Test_PushGroupReferences(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	logger := zap.NewNop()
+
+	g, _, err := weshnet.NewGroupMultiMember()
+	assert.NoError(t, err)
+
+	acc := cryptoutil.NewDeviceKeystore(keystore.NewMemKeystore(), nil)
+
+	omd, err := acc.MemberDeviceForGroup(g)
+	assert.NoError(t, err)
+
+	devicePK, err := omd.Public().Device.Raw()
+	assert.NoError(t, err)
+
+	ds, err := cryptoutil.NewDeviceSecret()
+	assert.NoError(t, err)
+
+	mkh, cleanup := cryptoutil.NewInMemMessageKeystore(logger)
+	defer cleanup()
+
+	// test with the ds counter
+	updateAndTestPushGroupReferences(ctx, mkh, devicePK, ds.Counter, g, t)
+
+	// do the same test with a new device secret counter
+	// so we can test if old references are deleted
+	updateAndTestPushGroupReferences(ctx, mkh, devicePK, ds.Counter+10, g, t)
```

Comment:
What is the expected behaviour with an out of bounds value such as `ds.Counter + precomputePushRefsCount + 1`? Can you test it?

Reply:
yes, I already did that at the end of this test, lines 584 and 593.